tests/schema/tests.py

```diff
@@ -1438,6 +1482,47 @@ class Meta:
         )
         self.assertIn("field_id", self.get_uniques(RelationModel._meta.db_table))
 
+    @isolate_apps("schema")
+    @unittest.skipUnless(connection.vendor == "postgresql", "PostgreSQL specific")
+    @skipUnlessDBFeature("supports_collation_on_charfield")
+    def test_relation_to_deterministic_collation_charfield(self):
+        deterministic_collation = connection.features.test_collations["deterministic"]
+
+        class CharModel(Model):
+            field = CharField(db_collation=deterministic_collation, unique=True)
+
+            class Meta:
+                app_label = "schema"
+
+        class RelationModel(Model):
+            field = OneToOneField(CharModel, CASCADE, to_field="field")
+
+            class Meta:
+                app_label = "schema"
+
+        # Create the table.
+        with connection.schema_editor() as editor:
+            editor.create_model(CharModel)
+            editor.create_model(RelationModel)
+        self.isolated_local_models = [CharModel, RelationModel]
+        constraints = self.get_constraints_for_column(CharModel, "field")
+        for constraint in constraints:
+            if constraint.startswith("schema_charmodel_field_"):
+                opclasses = self.get_constraint_opclasses(constraint)
+                self.assertIn("varchar_pattern_ops", opclasses)
+                break
+        else:
+            raise AssertionError("Constraint not found")
```

Comment:
I am not sure if we should also check for `varchar_pattern_ops` on `RelationModel.field_id`. At the moment it doesn't seem to have it, though it does have the collation. I am not sure if it should 🤔

Reply:
It exists for me: `schema_relationmodel_field_id_395fbb08_like ['varchar_pattern_ops']`.

Reply:
I pushed edits to tests.